src/java.base/share/classes/sun/nio/ch/SelectorImpl.java

```diff
@@ -225,6 +226,8 @@ protected final SelectionKey register(AbstractSelectableChannel ch,
             keys.remove(k);
             k.cancel();
             throw e;
+        } catch (CancelledKeyException ignored) {
+            keys.remove(k);
```

Comment:
I don't think keys.remove(k) should be here. If the key has been cancelled then it will be removed from the key set when the dereg queue is processed. Did you run into a test failure that lead to adding this?

Reply:
No failure, it was for symmetry with the other catch block; will remove.

Reply:
> Thanks

I will remove the `remove()` in the next commit.

Reply:
Removed in 925336449bfda90791b7e289d0008da4a9bc87a6.